File: src/1261-Find-Elements-in-a-Contaminated-Binary-Tree/1261.java
class FindElements {
    public FindElements(TreeNode root) {
        root.val = 0;
        data.add(0);
        dfs(root);
    }
    
    private Set<Integer> data = new HashSet();
    
    private void dfs(TreeNode root) {
        if (root.left != null) {
            root.left.val = root.val*2 + 1;
            data.add(root.left.val);
            dfs(root.left);
        }
        if (root.right != null) {
            root.right.val = root.val*2 + 2;
            data.add(root.right.val);
            dfs(root.right);
        }
    }
    
    public boolean find(int target) {
        return data.contains(target);
    }
}

File: src/1261-Find-Elements-in-a-Contaminated-Binary-Tree/1261.js
/**
 * @param {TreeNode} root
 */
var FindElements = function(root) {
    this.data = new Set();
    root.val = 0;
    this.data.add(0);
    
    let dfs = node => {
        if (node.left) {
            node.left.val = node.val * 2 + 1;
            this.data.add(node.left.val);
            dfs(node.left);
        }
        if (node.right) {
            node.right.val = node.val * 2 + 2;
            this.data.add(node.right.val);
            dfs(node.right);
        }
    }
    dfs(root);
};

/** 
 * @param {number} target
 * @return {boolean}
 */
FindElements.prototype.find = function(target) {
    return this.data.has(target);
};